core/src/main/java/org/testcontainers/containers/ContainerState.java

```diff
@@ -264,20 +268,27 @@ default void copyFileToContainer(Transferable transferable, String containerPath
             throw new IllegalStateException("copyFileToContainer can only be used with created / running container");
         }
 
-        try (
-            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-            TarArchiveOutputStream tarArchive = new TarArchiveOutputStream(byteArrayOutputStream)
-        ) {
-            tarArchive.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
-
-            transferable.transferTo(tarArchive, containerPath);
-            tarArchive.finish();
+        Path tarFile = Files.createTempFile("test-containers-copy-file-to-container", ".tar");
```

Comment:
Perhaps it is better to use `PipedInputStream`

Reply:
Docker Java creates a temp file anyway, so we are not winning much.

Reply:
that is something I am willing to fix in `docker-java` as well :)

Reply:
Shall we change this here now, so that we can avoid temp file creation altogether once docker-java is changed?